src/functions-and-arrays.js

```diff
@@ -1,20 +1,80 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1,num2) {
+  if (num1 > num2) { 
+    return num1;
+} else {
+  return num2;
+}
+} 
+
+
 // Iteration #2: Find longest word
+
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+// let findLongestWord = (array) => {
+//   let longest="";
+//   for(let i=0; i< array.length; i++){
+//       let nexti= i+1;
+//       if (array[i].length > array[nexti].length){
+// return(array[i]);
+// console.log(array[nexti].length);
+//       }else if (array[nexti].length > array[i].length){
+//           console.log(array[nexti]);
+ 
+//       }
+//   }
+//   return;
+// };
+// console.log(findLongestWord(words));
+
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(ArrayOfnumbers) {
+  let sum = 0;
+  ArrayOfnumbers.forEach (function(ArrayOfnumbers) {
+    sum += ArrayOfnumbers;
+  });
+  return sum;
+}
+
+
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+let total = 0;
+for(let i = 0; i < numbersAvg.length; i++) {
+    total += numbersAvg[i];
+}
+let avg = total / numbersAvg.length;
```

Comment:
You may have noticed that all your Jasmine tests are still red. The reason is this code block here.

The problem here is that you are not actually creating a function. If you placed these operations in a function, then the Jasmine test suite should run as it should.

```suggestion
const numbers = [2, 6, 9 ,10, 7 ,4 ,1 , 9];

function numbersAvg(arrOfNumbers) {
  let total = 0;
  for(let i = 0; i < arrOfNumbers.length; i++) {
      total += arrOfNumbers[i];
  }
  
  let avg = total / arrOfNumbers.length;
  
  // return some value....
}

```